src/forms/components/encounter/ohri-encounter-form.tsx

```diff
@@ -113,12 +123,20 @@ export const OHRIEncounterForm: React.FC<OHRIEncounterFormProps> = ({
       setEncounterLocation(encounter.location);
     } else {
       allFormFields.forEach(field => {
-        if (field.questionOptions.rendering == 'checkbox') {
-          tempInitVals[field.id] = [];
-        } else if (field.questionOptions.rendering == 'toggle') {
-          tempInitVals[field.id] = false;
+        let value = null;
+        if (!isEmpty(field.questionOptions.defaultValue)) {
+          value = inferInitialValueFromDefaultFieldValue(field, encounterContext);
```

Comment:
Why don't you pass the handler here? You have the `field` and you can get the field type, or it is not possible at this point? I don't like the ideia of getting the handler inside of the `inferInitialValueFromDefaultFieldValue`. The method is doing more than it is supposed to

Reply:
Resolved